new page for LBM based CFD codes and packages

nileshsawant opened the pull request:
Covers M-Star and marbles. Mentions Ansys Discovery. Thank you for the checks.

Commit:
new page for LBM based CFD codes and packages

yandthj commented on docs/Documentation/Applications/LBMcfd.md:
```diff
@@ -0,0 +1,124 @@
+parent: Applications
+---
+
+# Lattice Boltzmann Method based CFD applications: M-Star, marbles, Ansys Discovery
```
```suggestion
# Lattice Boltzmann Method based CFD Applications: M-Star, marbles, Ansys Discovery
```

yandthj commented on docs/Documentation/Applications/LBMcfd.md:
```diff
@@ -0,0 +1,124 @@
+-->
+
+
+The lattice Boltzmann method (LBM) is a novel and unique simulation method whose meshless and exact conservation combines the best parts of the immersed boundary method and the finite volume method, without the hassle of mesh generation. The method completely does away with spatial discretisation in favour of advection of particles with pre-determined discrete velocities such that they hop to lat ...
```
```suggestion
The Lattice Boltzmann Method (LBM) is a novel and unique simulation method whose meshless and exact conservation combines the best parts of the immersed boundary method and the finite volume method, without the hassle of mesh generation. The method completely does away with spatial discretisation in favour of advection of particles with pre-determined discrete velocities such that they hop to lattice sites on a structured grid. Consequently, the method does not incur any errors due to numerical dissipation or dispersion, thereby ensuring conservation of globally conserved properties like mass and total energy. The time discretisation is done with the trapezoidal rule, which results into transient simualations with second order accuracy in time. The LBM has been successfully applied to a range of problems in fluid dynamics including but not limited to transitional flows, flows involving complex moving geometries, compressible flows, multiphase flows, rarefied gases, combustion, electrochemical devices etc.. 
```

yandthj replied:
I don't think we need this much explanation on the method, as that's not really the purpose of our documentation. A brief intro is okay. This is information that should be covered by the application's own documentation.

yandthj commented on docs/Documentation/Applications/LBMcfd.md:
```diff
@@ -0,0 +1,124 @@
+| [NREL marbles](https://nrel.github.io/marbles/VandV.html)                         | y      | y       | y           | y      | y      | Free | Open source, compressible  |  
+| [Ansys Discovery](https://www.ansys.com/products/3d-design/ansys-discovery)  | y      | x       | x           | x      | y      | $    | GUI, geometry modeling       |
+
+Only M-Star and marbles can be run on the Kestrel HPC system. Users with an access to GPU enabled Windows computers or Virtual Machines may try Ansys Discovery at their own discretion.
```
Since Ansys Discovery isn't available on our systems, I don't think we should include it in the documentation. The documentation is intended to be how to do things on HPC, but not necessarily explaining scientific content in general. Message me if you want to discuss more.

yandthj commented on docs/Documentation/Applications/LBMcfd.md:
```diff
@@ -0,0 +1,124 @@
+
+Only M-Star and marbles can be run on the Kestrel HPC system. Users with an access to GPU enabled Windows computers or Virtual Machines may try Ansys Discovery at their own discretion.
+
+## Installation and usage on Kestrel
```
```suggestion
## Installation and Usage on Kestrel
```

yandthj commented on docs/Documentation/Applications/LBMcfd.md:
```diff
@@ -0,0 +1,124 @@
+
+### NREL marbles
+
+Source code of marbles is available on github. It can be compiled and run on nvidia and AMD GPUs as well as Intel, AMD and Apple M series CPUs. Here, we show the process to compile and run it on the Kestrel HPC system with nvidia GPUs.
```
```suggestion
The marbles source code is available on Github. It can be compiled and run on NVIDIA and AMD GPUs as well as Intel, AMD and Apple M series CPUs. Here, we show the process to compile and run it on the Kestrel HPC system with NVIDIA GPUs.
```

yandthj replied:
Can you link the github repo here?

yandthj replied:
It looks like marbles is capitalized as "MARBLES" in it's documentation that you linked. Can you update this documentation to use that convention too?

yandthj commented on docs/Documentation/Applications/LBMcfd.md:
````diff
@@ -0,0 +1,124 @@
+
+The test case for flow through fractures with heated isothermal walls can be tried out as follows:
+```
+$ salloc -A hpcapps -t 00:30:00 --nodes=1 --ntasks-per-node=32 --mem=80G --gres=gpu:1 --partition=debug
````
```suggestion
$ salloc -A <allocation handle> -t 00:30:00 --nodes=1 --ntasks-per-node=32 --mem=80G --gres=gpu:1 --partition=debug
```

yandthj commented on docs/Documentation/Applications/LBMcfd.md:
```diff
@@ -0,0 +1,124 @@
+![Velocity Cracks Demo](LBMcfd_metadata/velocity_cracks.gif)
+*Animation credit: [@eyoung55](https://github.com/eyoung55)*
+
+[marbles](https://nrel.github.io/marbles/VandV.html) is an in-house effort to make a free lattice Boltzmann solver available to the community. We encourage users to contact us for help setting up your problem or to request additional features. Please visit the repository and create a [`New issue`](https://github.com/NREL/marbles/issues) or [email](mailto:nsawant@nrel.gov) us directly. A [machine  ...
```
```suggestion
[marbles](https://nrel.github.io/marbles/VandV.html) is an in-house effort to make a free Lattice Boltzmann solver available to the community. We encourage users to contact us for help setting up your problem or to request additional features. Please visit the repository and create a [New issue](https://github.com/NREL/marbles/issues) or [email](mailto:nsawant@nrel.gov) us directly. A [machine learning framework](https://github.com/nileshsawant/mlForLBM) for using marbles in the loop as a data generator has also been created. Pre-built executibles, `marbles3d.gnu.x86-milan.TPROF.MPI.ex` and `marbles3d.gnu.TPROF.MPI.CUDA.ex`, can also be made available on request.
```

yandthj commented on docs/Documentation/Applications/LBMcfd.md:
```diff
@@ -0,0 +1,124 @@
+### M-Star
+
+!!! Warning
+	Users are advised to not leave this application open when not working on it. The license of this product allows only one user at a time.
```
```suggestion
	Please close the application when not actively using it. The license of this product allows only one user at a time.
```

yandthj commented on docs/Documentation/Applications/LBMcfd.md:
```diff
@@ -0,0 +1,124 @@
+!!! Warning
+	Users are advised to not leave this application open when not working on it. The license of this product allows only one user at a time.
+
+The M-Star GUI can be accessed though a [FastX virtual desktop](../Viz_Analytics/virtualgl_fastx.md). M-Star is a resourse intensive application whose backed uses cuda aware openmpi to utlise multiple GPUs for computation. The application is always executed on a dedicated compute node while its GUI is interacted with on a FastX node. The steps to use M-Star are as follows:
```
```suggestion
The M-Star GUI can be accessed though a [FastX virtual desktop](../Viz_Analytics/virtualgl_fastx.md). M-Star is a resource intensive application whose backend uses Cuda aware OpenMPI to utilize multiple GPUs for computation. The application should always be run on a dedicated compute node while you can interact with the GUI through a FastX session. The steps to use M-Star are as follows:
```

yandthj commented on docs/Documentation/Applications/LBMcfd.md:
````diff
@@ -0,0 +1,124 @@
+```
+$ salloc -A <projectname> -t 01:00:00 --nodes=1 --ntasks-per-node=64 --mem=160G --gres=gpu:2 --partition=debug
+```
+2. Wait until you obtain an allocation, The terminal will display `<username>@<nodename>` when successful.
````
```suggestion
2. Wait until you obtain an allocation. The terminal will display `<username>@<nodename>` when successful.
```

yandthj commented on docs/Documentation/Applications/LBMcfd.md:
```diff
@@ -0,0 +1,124 @@
+---
+title: LBM-cfd
```
```suggestion
title: LBM CFD
```

yandthj commented on mkdocs.yml:
```diff
@@ -88,6 +88,7 @@ nav:
           - IDAES Solvers : Documentation/Applications/idaes_solvers.md
           - IPOPT:      Documentation/Applications/ipopt.md
           - LAMMPS:     Documentation/Applications/lammps.md
+          - LBM CFD:     Documentation/Applications/LBMcfd.md
```
Can you add an entry to the Applications index page table?

yandthj commented on docs/Documentation/Applications/LBMcfd.md:
````diff
@@ -0,0 +1,124 @@
+$ srun -n 1 marbles3d.gnu.TPROF.MPI.CUDA.ex isothermal_cracks.inp
+```
+
+Results can be viewed in [ParaView](https://nrel.github.io/HPC/Documentation/Viz_Analytics/paraview/) or [VisIT](https://nrel.github.io/HPC/Documentation/Viz_Analytics/visit/).
````
Can you use the file paths for the links instead of urls?

nileshsawant committed:
Update docs/Documentation/Applications/LBMcfd.md

Co-authored-by: Haley Yandt <46908710+yandthj@users.noreply.github.com>

nileshsawant committed:
Update docs/Documentation/Applications/LBMcfd.md

Co-authored-by: Haley Yandt <46908710+yandthj@users.noreply.github.com>

nileshsawant committed:
Update docs/Documentation/Applications/LBMcfd.md

Co-authored-by: Haley Yandt <46908710+yandthj@users.noreply.github.com>

nileshsawant committed:
Update docs/Documentation/Applications/LBMcfd.md

Co-authored-by: Haley Yandt <46908710+yandthj@users.noreply.github.com>

nileshsawant committed:
Update docs/Documentation/Applications/LBMcfd.md

Co-authored-by: Haley Yandt <46908710+yandthj@users.noreply.github.com>

nileshsawant committed:
Update docs/Documentation/Applications/LBMcfd.md

Co-authored-by: Haley Yandt <46908710+yandthj@users.noreply.github.com>

nileshsawant committed:
Update docs/Documentation/Applications/LBMcfd.md

Co-authored-by: Haley Yandt <46908710+yandthj@users.noreply.github.com>

nileshsawant committed:
Update docs/Documentation/Applications/LBMcfd.md

Co-authored-by: Haley Yandt <46908710+yandthj@users.noreply.github.com>

nileshsawant committed:
Update docs/Documentation/Applications/LBMcfd.md

Co-authored-by: Haley Yandt <46908710+yandthj@users.noreply.github.com>

nileshsawant committed:
Update docs/Documentation/Applications/LBMcfd.md

Co-authored-by: Haley Yandt <46908710+yandthj@users.noreply.github.com>

nileshsawant committed:
Correct capitalization and enhance LBM description

Updated capitalization of 'MARBLES' and improved clarity in the description of the Lattice Boltzmann Method. Reduced method description.

nileshsawant committed:
Fix links to visualization tools in LBMcfd.md

Updated links to ParaView and VisIT documentation.

nileshsawant committed:
Removed ansys discovery

Removed ansys discovery

yandthj commented on docs/Documentation/Applications/LBMcfd.md:
```diff
@@ -0,0 +1,123 @@
+![Velocity Cracks Demo](LBMcfd_metadata/velocity_cracks.gif)
+*Animation credit: [@eyoung55](https://github.com/eyoung55)*
+
+[marbles](https://nrel.github.io/marbles/VandV.html) is an in-house effort to make a free Lattice Boltzmann solver available to the community. We encourage users to contact us for help setting up your problem or to request additional features. Please visit the repository and create a [New issue](https://github.com/NREL/marbles/issues) or [email](mailto:nsawant@nrel.gov) us directly. A [machine le ...
```
```suggestion
[MARBLES](https://nrel.github.io/marbles/VandV.html) is an in-house effort to make a free Lattice Boltzmann solver available to the community. We encourage users to contact us for help setting up your problem or to request additional features. Please visit the repository and create a [New issue](https://github.com/NREL/marbles/issues) or [email](mailto:nsawant@nrel.gov) us directly. A [machine learning framework](https://github.com/nileshsawant/mlForLBM) for using MARBLES in the loop as a data generator has also been created. Pre-built executibles, `marbles3d.gnu.x86-milan.TPROF.MPI.ex` and `marbles3d.gnu.TPROF.MPI.CUDA.ex`, can also be made available on request.
```

yandthj commented on docs/Documentation/Applications/LBMcfd.md:
```diff
@@ -0,0 +1,123 @@
+
+### NREL MARBLES
+
+The MARBLES source code is available on Github. It can be compiled and run on NVIDIA and AMD GPUs as well as Intel, AMD and Apple M series CPUs. Here, we show the process to compile and run it on the Kestrel HPC system with NVIDIA GPUs.
```
Can you link to the source code here

yandthj commented on docs/Documentation/Applications/LBMcfd.md:
```diff
@@ -0,0 +1,123 @@
+
+The Lattice Boltzmann Method (LBM) is a novel and unique simulation method whose meshless algorithm and exact conservation combines the best parts of the immersed boundary method and the finite volume method, without the hassle of mesh generation. The method completely does away with spatial discretisation in favour of advection of particles with pre-determined discrete velocities such that they  ...
+
+Its meshfree nature makes it very convenient to handle and resolve complex geomtries such as cracks and porous microstructures. The algorithmically simple nature of the LBM which consists of hopping of particles by a pre-determined distance followed by a local update for time increment makes the solver trivial to implement on a GPU, which results into very fast and scalable solvers that can be us ...
```
```suggestion
Its meshfree nature makes it very convenient to handle and resolve complex geometries such as cracks and porous microstructures. The algorithmically simple nature of the LBM, which consists of a hopping of particles by a pre-determined distance followed by a local update for a time increment, makes the solver trivial to implement on a GPU, which results into very fast and scalable solvers that can be used as data generators in a machine learning pipeline. 
```

nileshsawant committed:
Update docs/Documentation/Applications/LBMcfd.md

Co-authored-by: Haley Yandt <46908710+yandthj@users.noreply.github.com>

nileshsawant committed:
Update docs/Documentation/Applications/LBMcfd.md

Co-authored-by: Haley Yandt <46908710+yandthj@users.noreply.github.com>

nileshsawant committed:
Update MARBLES source code reference in documentation

Added a link to the MARBLES source code repository.
